lightning-net-tokio/src/lib.rs

```diff
@@ -218,11 +202,10 @@ impl Connection {
 		}
 		if let Disconnect::PeerDisconnected = disconnect_type {
 			peer_manager_ref.socket_disconnected(&our_descriptor);
-			Self::event_trigger(&mut us.lock().unwrap());
```

Comment:
Oops, I forgot that we're also relying on the event trigger thing here to call `peer_manager.process_events()` (which is also dumb...). Can you add a call to it after the select in the loop and maybe here after the disconnect as well? (while you're at it, we don't need the `peer_manager_ref` clone anymore, it seems?).